### ISSUE TITLE
InvalidArrayOffset is not emitted for array<mixed,T> or T[]

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#964]
Expected: Both functions should warn about using `stdClass` as an array offset
Observed: Only the latter warns about this.

Is this by design?

```php
<?php

/**
 * @param array<mixed,string> $y (this doesn't warn)
 */
function example($y) : void {
    $o = new stdClass();
    echo $y[$o];
}

/**
 * @param array<int|string,string> $x (This properly warns about invalid array offsets)
 */
function example2($x) : void {
    $o = new stdClass();
    echo $x[$o];
}
```

See https://getpsalm.org/r/b65ff627af

**Comment by muglug** [comment]
Definitely not by design, definitely by bug.
